activerecord/test/cases/adapter_test.rb

```diff
@@ -296,5 +279,22 @@ def setup
         assert !@connection.transaction_open?
       end
     end
+
+    def test_disable_referential_integrity
```

Comment:
Because we can't use `disable_referential_integrity_with_alter_constraint` within transaction.

```sql
-- setup
DROP TABLE "fk_test_has_fk";
CREATE TABLE "fk_test_has_fk" ("fk_id" integer NOT NULL);
DROP TABLE "fk_test_has_pk";
CREATE TABLE "fk_test_has_pk" ("pk_id" serial primary key);
ALTER TABLE "fk_test_has_fk" ADD CONSTRAINT "fk_name" FOREIGN KEY ("fk_id") REFERENCES "fk_test_has_pk" ("pk_id");

-- test
BEGIN;

ALTER TABLE fk_test_has_fk ALTER CONSTRAINT fk_name DEFERRABLE;

SET CONSTRAINTS ALL DEFERRED;

INSERT INTO fk_test_has_fk VALUES(1);
INSERT INTO fk_test_has_pk VALUES(1);

ALTER TABLE fk_test_has_fk ALTER CONSTRAINT fk_name NOT DEFERRABLE;
-- ERROR:  cannot ALTER TABLE "fk_test_has_fk" because it has pending trigger events
```

So we must put this test within `AdapterTestWithoutTransaction`